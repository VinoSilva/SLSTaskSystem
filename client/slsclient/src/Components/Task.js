import React, { Component } from "react";
import { connect } from "react-redux";
import { showAddTask, hideAddTask } from "../actions/taskAddAction";

function mapStateToProps(state) {
  return {
    isAddForm: state.taskFormReducer.isAddForm
  };
}

// // in this object, keys become prop names,
// // and values should be action creator functions.
// // They get bound to `dispatch`.
// const mapDispatchToProps = dispatch => ({
//     showAddTask,
//     hideAddTask,
//     dispatch
// })

export class Task extends Component {
  constructor(props) {
    super(props);

    this.onClickAddTask = this.onClickAddTask.bind(this);
  }

  onClickAddTask() {
    if (this.props.isAddForm) {
      //Hide the form
      this.props.dispatch(hideAddTask());
    } else {
      //Show the form
      this.props.dispatch(showAddTask());
    }
  }

  renderAddTaskForm() {
    if (this.props.isAddForm) {
      return (
        <div id="addTaskForm">
          <h1>Add Form</h1>
        </div>
      );
    }

    return <div></div>;
  }


  render() {
    return (
      <div>
        <h1 id="taskHeader">Task List</h1>
        <div id="taskList"></div>

        {this.renderAddTaskForm()}

        <button
          className="btn btn-info"
          id="addTaskBtn"
          onClick={this.onClickAddTask}
        >
          Add Form
        </button>
      </div>
    );
  }
}

// export default connect(mapStateToProps,mapDispatchToProps)(Counter);
// export default  connect(mapDispatchToProps,mapStateToProps)(Task);
export default connect(mapStateToProps)(Task);
