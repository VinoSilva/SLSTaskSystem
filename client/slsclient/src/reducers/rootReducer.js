import {combineReducers} from "redux";

import taskReducer from './taskReducer';
import taskFormReducer from './taskFormReducer';
import taskSearchReducer from './taskSearchReducer';
import taskUpdateReducer from './taskUpdateReducer';

export default combineReducers({
    taskReducer,
    taskFormReducer,
    taskSearchReducer,
    taskUpdateReducer
});