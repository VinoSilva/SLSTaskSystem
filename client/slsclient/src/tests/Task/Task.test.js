import React from "react";
import ReactDOM from "react-dom";
import Task from "../../Components/Task";

import { mount } from "enzyme";

import renderer from "react-test-renderer";

import thunk from 'redux-thunk';
import { createStore,applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../../reducers/rootReducer'; 

let store;

describe("Task component", () => {
  
  beforeEach(() => {
    store = createStore(rootReducer,applyMiddleware(thunk)); 
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    const render = ReactDOM.render(<Provider store = {store}> <Task /> </Provider>,  div);
    
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Task matches the snapshot", () => {
    const tree = renderer.create(<Provider store = {store}> <Task /> </Provider>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Task Component should have a div with id "taskList"', () => {
    const wrapper = mount(<Provider store = {store}><Task /></Provider>);
    
    const taskList = wrapper.find("#taskList");
    expect(taskList).toBeTruthy();
    expect(taskList.type()).toBe("div");
  });

  it('Task Component should have a button with id "addTaskBtn"', () => {
    const wrapper = mount(<Provider store = {store}><Task /></Provider>);

    const addTaskBtn = wrapper.find("#addTaskBtn");
    expect(addTaskBtn).toBeTruthy();
    expect(addTaskBtn.type()).toBe("button");
  });

  it('Task Component should have a button with id "addTaskBtn"', () => {
    const wrapper = mount(<Provider store = {store}><Task /></Provider>);

    const addTaskBtn = wrapper.find("#addTaskBtn");
    addTaskBtn.simulate('click');

    const addTaskForm = wrapper.find('#addTaskForm');

    expect(addTaskForm).toBeTruthy();

    expect(addTaskForm.type()).toBe("div");
  });
});
