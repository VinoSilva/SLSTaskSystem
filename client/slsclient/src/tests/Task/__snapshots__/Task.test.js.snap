// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component Task matches the snapshot 1`] = `
Array [
  " ",
  <div>
    <h1
      id="taskHeader"
    >
      Task List
    </h1>
    <div
      id="taskList"
    />
    <div />
    <button
      className="btn btn-info"
      id="addTaskBtn"
      onClick={[Function]}
    >
      Add Form
    </button>
  </div>,
  " ",
]
`;
